sdk/src/test/scala/com/horizen/account/block/AccountBlockTest.scala

```diff
@@ -0,0 +1,941 @@
+
+  val seed: Long = 11L
+  val parentId: ModifierId = getRandomBlockId(seed)
+
+  val generatedDataSeed = 908
+  val vrfGenerationPrefix = "SidechainBlockTest"
+
+  if (false) {
+    VrfGeneratedDataProvider.updateVrfSecretKey(vrfGenerationPrefix, generatedDataSeed)
+    VrfGeneratedDataProvider.updateVrfProof(vrfGenerationPrefix, generatedDataSeed)
+  }
+
+  val vrfKeyPair: Option[(VrfSecretKey, VrfPublicKey)] = {
+    val secret: VrfSecretKey = VrfGeneratedDataProvider.getVrfSecretKey(vrfGenerationPrefix, generatedDataSeed)
+    val publicKey: VrfPublicKey = secret.publicImage()
+    Option((secret, publicKey))
+  }
+
+  val (accountPayment, forgerMetadata) = ForgerAccountFixture.generateForgerAccountData(seed, vrfKeyPair)
+  val vrfProof: VrfProof = VrfGeneratedDataProvider.getVrfProof(vrfGenerationPrefix, generatedDataSeed)
+
+  // Create Block with Txs, MainchainBlockReferencesData, MainchainHeaders and Ommers
+  // Note: block is semantically invalid because Block contains the same MC chain as Ommers, but it's ok for serialization test
+  val block: AccountBlock = createBlock(
+    sidechainTransactions = Seq(
+      getEoa2EoaLegacyTransaction,
+      getUnsignedEoa2EoaLegacyTransaction
+    ),
+    receipts = Seq(
+      new EthereumConsensusDataReceipt(2, ReceiptStatus.SUCCESSFUL.id, BigInteger.valueOf(112233), Array.empty[EvmLog]),
+      new EthereumConsensusDataReceipt(2, ReceiptStatus.FAILED.id, BigInteger.valueOf(22334455), Array.empty[EvmLog])
+    ),
+    mainchainBlockReferencesData = Seq(mcBlockRef1.data, mcBlockRef2.data),
+    mainchainHeaders = Seq(mcBlockRef2.header, mcBlockRef3.header, mcBlockRef4.header),
+    ommers = generateOmmersSeq(parentId, 123444L,
+      Seq(
+        (Seq(mcBlockRef1.data), Seq(mcBlockRef1.header, mcBlockRef2.header)),
+        (Seq(), Seq()),
+        (Seq(mcBlockRef2.data, mcBlockRef3.data, mcBlockRef4.data), Seq(mcBlockRef3.header, mcBlockRef4.header))
+      ),
+      rnd = random
+    ),
+    rnd = random
+  )
+
+  @Test
+  def testToJson(): Unit = {
+    val sb = createBlock()
+
+    val serializer = ApplicationJsonSerializer.getInstance()
+    serializer.setDefaultConfiguration()
+
+    val jsonStr = serializer.serialize(sb)
+
+    val node : JsonNode = serializer.getObjectMapper.readTree(jsonStr)
+
+    try {
+      val id = node.path("id").asText()
+      assertEquals("Block id json value must be the same.",
+        BytesUtils.toHexString(idToBytes(sb.id)), id)
+    }catch {
+      case _: Throwable => fail("Block id doesn't not found in json.")
+    }
+    try {
+      val parentId = node.path("parentId").asText()
+      assertEquals("Block parentId json value must be the same.",
+        BytesUtils.toHexString(idToBytes(sb.parentId)), parentId)
+    }catch {
+      case _: Throwable => fail("Block parentId doesn't not found in json.")
```

Comment:
This will catch also the assertionerror thrown by assertEquals, masking the original error. Consider to catch just Exception and not Throwable or to remove the try/catch entirely. Same below

Reply:
Done